OSDOCS-21603: Add 4.22.10 z-stream release notes (#118197)
openshift/openshift-docs: modules/zstream-4-22-10.adoc

// Module included in the following assemblies:
//
// * release_notes/ocp-4-22-release-notes.adoc

:_mod-docs-content-type: REFERENCE
[id="zstream-4-22-10_{context}"]
= RHSA-2026:54770 - {product-title} {product-version}.10 bug fix and security update

Issued: 18 August 2026

[role="_abstract"]
{product-title} release {product-version}.10 is now available. The list of fixed issues that are included in the update is documented in the link:https://access.redhat.com/errata/RHSA-2026:54770[RHSA-2026:54770] advisory. The RPM packages that are included in the update are provided by the link:https://access.redhat.com/errata/RHBA-2026:54768[RHBA-2026:54768] advisory.

Space precluded documenting all of the container images for this release in the advisory.

You can view the container images in this release by running the following command:

[source,terminal]
----
$ oc adm release info 4.22.10 --pullspecs
----

[id="zstream-4-22-10-enhancements_{context}"]
== Enhancements

> * On Azure {product-title} {hcp} (HCP) clusters, worker nodes had no mechanism to authenticate to Azure Container Registry (ACR) by using managed identities because the `HostedCluster` API lacked support for configuring ACR credentials with a user-assigned managed identity. Image pulls from ACR on worker nodes failed with authentication errors. Customers had to manually configure image pull secrets in every namespace that needed to pull from ACR, which was operationally burdensome and did not scale. An optional `containerRegistry` field was added to `AzurePlatformSpec` that accepts a user-assigned managed identity. When configured, the Control Plane Operator writes the user-assigned identity ID into the worker `cloud.conf` ConfigMap, and the `NodePool` controller attaches the managed identity to the worker virtual machine scale sets by using {cap-azure-short}. When `containerRegistry` is configured with a user-assigned managed identity, the kubelet ACR credential provider authenticates automatically without requiring image pull secrets. (link:https://redhat.atlassian.net/browse/OCPBUGS-94518[OCPBUGS-94518])

Reviewer: Enhancements should start with the now and be in active voice:

You can now authenticate worker nodes to...

Reviewer: Word count for some sentences exceeds 22 words. Consider pruning.

> * On Azure {product-title} {hcp} (HCP) clusters, worker nodes had no mechanism to authenticate to Azure Container Registry (ACR) by using managed identities because the `HostedCluster` API lacked support for configuring ACR credentials with a user-assigned managed identity. Image pulls from ACR on worker nodes failed with authentication errors. Customers had to manually configure image pull secrets in every namespace that needed to pull from ACR, which was operationally burdensome and did not scale. An optional `containerRegistry` field was added to `AzurePlatformSpec` that accepts a user-assigned managed identity. When configured, the Control Plane Operator writes the user-assigned identity ID into the worker `cloud.conf` ConfigMap, and the `NodePool` controller attaches the managed identity to the worker virtual machine scale sets by using {cap-azure-short}. When `containerRegistry` is configured with a user-assigned managed identity, the kubelet ACR credential provider authenticates automatically without requiring image pull secrets. (link:https://redhat.atlassian.net/browse/OCPBUGS-94518[OCPBUGS-94518])

Reviewer: ", which was operationally burdensome and did not scale" needed? Could prune.

> * On Azure Red Hat OpenShift hosted control plane (HCP) clusters, worker nodes had no mechanism to authenticate to Azure Container Registry (ACR) by using managed identities because the `HostedCluster` API lacked support for configuring ACR credentials with a user-assigned managed identity. Image pulls from ACR on worker nodes failed with authentication errors. Customers had to manually configure image pull secrets in every namespace that needed to pull from ACR, which was operationally burdensome and did not scale. An optional `containerRegistry` field was added to `AzurePlatformSpec` that accepts a user-assigned managed identity. When configured, the Control Plane Operator writes the user-assigned identity ID into the worker `cloud.conf` ConfigMap, and the `NodePool` controller attaches the managed identity to the worker virtual machine scale sets by using {cap-azure-short}. When `containerRegistry` is configured with a user-assigned managed identity, the kubelet ACR credential provider authenticates automatically without requiring image pull secrets. (link:https://redhat.atlassian.net/browse/OCPBUGS-94518[OCPBUGS-94518])

Reviewer 2: ```suggestion
* On Azure {product-title} {hcp} (HCP) clusters, worker nodes had no mechanism to authenticate to Azure Container Registry (ACR) by using managed identities because the `HostedCluster` API lacked support for configuring ACR credentials with a user-assigned managed identity. Image pulls from ACR on worker nodes failed with authentication errors. Customers had to manually configure image pull secrets in every namespace that needed to pull from ACR, which was operationally burdensome and did not scale. An optional `containerRegistry` field was added to `AzurePlatformSpec` that accepts a user-assigned managed identity. When configured, the Control Plane Operator writes the user-assigned identity ID into the worker `cloud.conf` ConfigMap, and the `NodePool` controller attaches the managed identity to the worker virtual machine scale sets by using {cap-azure-short}. When `containerRegistry` is configured with a user-assigned managed identity, the kubelet ACR credential provider authenticates automatically without requiring image pull secrets. (link:https://redhat.atlassian.net/browse/OCPBUGS-94518[OCPBUGS-94518])
```

[id="zstream-4-22-10-fixed-issues_{context}"]
== Fixed issues

> * Before this update, when you changed a `ProvisioningRequest` on day 2 to update non-hardware fields such as `clusterInstance` parameter labels, annotations, or policy parameters, the `ProvisioningRequest` controller used the resource generation as the `NodeAllocationRequest` `ConfigTransactionId` and updated the `NodeAllocationRequest` spec whenever that ID changed. As a consequence, the `NodeAllocationRequest` controller acknowledged the change without updating `ObservedConfigTransactionId`, so the `ProvisioningRequest` could remain stuck in `Hardware provisioning completed: Created` with the log message `Waiting for NodeAllocationRequest to be configured`. With this release, the `ProvisioningRequest` controller no longer treats non-hardware spec changes as requiring `NodeAllocationRequest` reconfiguration. As a result, day-2 `ProvisioningRequest` updates that do not affect hardware provisioning complete without getting stuck. (link:https://redhat.atlassian.net/browse/OCPBUGS-90659[OCPBUGS-90659])

Reviewer: With this release versus With this update. Maybe the latter is more consistent?

> * Before this update, the NetworkPolicy in the `openshift-apiserver` namespace blocked traffic on the port used by the `check-endpoints` service. As a consequence, Prometheus could not scrape those metrics, and the `TargetDown` warning alert could fire when all `check-endpoints` targets were unreachable for more than 15 minutes. With this release, the NetworkPolicy allows traffic on the designated port for the `check-endpoints` service. As a result, Prometheus can reach the endpoints successfully, and the `TargetDown` alert no longer fires for this target. (link:https://redhat.atlassian.net/browse/OCPBUGS-100168[OCPBUGS-100168])

Reviewer 2: ```suggestion
* Before this update, the `NetworkPolicy` object in the `openshift-apiserver` namespace blocked traffic on the port used by the `check-endpoints` service. As a consequence, Prometheus could not scrape those metrics, and the `TargetDown` warning alert could fire when all `check-endpoints` targets were unreachable for more than 15 minutes. With this release, the `NetworkPolicy` object allows traffic on the designated port for the `check-endpoints` service. As a result, Prometheus can reach the endpoints successfully, and the `TargetDown` alert no longer fires for this target. (link:https://redhat.atlassian.net/browse/OCPBUGS-100168[OCPBUGS-100168])
```

* Before this update, when you added a new vSphere failure domain that used a MachineSet with a custom `providerSpec.Template` name, the Machine Config Operator boot image controller looked up the VM template only by its own computed name and ignored `providerSpec.Template`. As a consequence, reconciliation for that failure domain could fail, and a customer-managed VM with the same computed name outside the MCO workspace folder could be mistaken for the MCO template and overwritten. With this release, the controller checks `providerSpec.Template` first, falls back to the computed name only when the template is not found, creates the template from the OVA when needed, and leaves name matches outside `providerSpec.Workspace.Folder` untouched. As a result, new vSphere failure domains with custom template names reconcile successfully, and customer-managed VMs outside the MCO workspace folder are no longer at risk of being overwritten. (link:https://redhat.atlassian.net/browse/OCPBUGS-104563[OCPBUGS-104563])

> * Before this update, on a Telecom Boundary Clock (T-BC) PTP configuration, `ts2phc` could start before the upstream PTP source was stable enough and before `phc2sys` was ready. As a consequence, the T-BC took a long time to converge, and `phc2sys` could adjust the system clock too early based on an irrelevant `ts2phc` offset. With this release, `ts2phc` start on T-BC is delayed until the upstream source is qualified and `phc2sys` is ready; T-GM behavior is unchanged. As a result, T-BC no longer starts `ts2phc` against an unqualified upstream source. (link:https://redhat.atlassian.net/browse/OCPBUGS-104578[OCPBUGS-104578])

Reviewer 2: ```suggestion
* Before this update, on a Telecom Boundary Clock (T-BC) PTP configuration, the `ts2phc` system daemon could start before the upstream PTP source was stable enough and before `phc2sys` system clock sync was ready. As a consequence, the T-BC took a long time to converge, and `phc2sys` system clock sync could adjust the system clock too early based on an irrelevant `ts2phc` system daemon offset. With this release, the `ts2phc` system daemon start on T-BC is delayed until the upstream source is qualified and the `phc2sys` system clock is ready; T-GM behavior is unchanged. As a result, T-BC no longer starts the `ts2phc` system daemon against an unqualified upstream source. (link:https://redhat.atlassian.net/browse/OCPBUGS-104578[OCPBUGS-104578])
```

Reviewer 2: Please add a step in the Cursor command to check the repo for appropriate nouns that follow `objects/CR` etc.

> * Before this update, on a Telecom Boundary Clock (T-BC) PTP configuration, the `ts2phc` system daemon could start before the upstream PTP source was stable enough and before `phc2sys` system clock sync was ready. As a consequence, the T-BC took a long time to converge, and `phc2sys` system clock sync could adjust the system clock too early based on an irrelevant `ts2phc` system daemon offset. With this release, the `ts2phc` system daemon start on T-BC is delayed until the upstream source is qualified and the `phc2sys` system clock is ready; T-GM behavior is unchanged. As a result, T-BC no longer starts the `ts2phc` system daemon against an unqualified upstream source. (link:https://redhat.atlassian.net/browse/OCPBUGS-104578[OCPBUGS-104578])

Reviewer: Expand T-GM on first use?

Telecom Grandmaster ?

> * Before this update, when you viewed Quick Start `{{execute}}` code snippets in the {product-title} web console, leading whitespace in the code block could render as an empty line before the command text. As a consequence, execute snippets across Quick Starts could display an extra blank line above the command, making the content harder to read. With this release, leading whitespace rendering in Quick Start execute code snippets is corrected. As a result, execute code blocks display the command text without a spurious empty line above it. (link:https://redhat.atlassian.net/browse/OCPBUGS-105611[OCPBUGS-105611])

Reviewer 2: ```suggestion
* Before this update, when you viewed *Quick Start* `{{execute}}` code snippets in the {product-title} web console, leading whitespace in the code block could render as an empty line before the command text. As a consequence, execute snippets across *Quick Starts* could display an extra blank line above the command, making the content harder to read. With this release, leading whitespace rendering in *Quick Start* execute code snippets is corrected. As a result, execute code blocks display the command text without an empty line above it. (link:https://redhat.atlassian.net/browse/OCPBUGS-105611[OCPBUGS-105611])
```
